src/battle_setup.c

```diff
@@ -899,6 +899,11 @@ u8 GetTrainerBattleTransition(void)
     u32 trainerId = SanitizeTrainerId(gTrainerBattleOpponent_A);
     u32 trainerClass = GetTrainerClassFromId(gTrainerBattleOpponent_A);
 
+    #if BATTLE_FRONTIER_INTRO_FOR_BP_FIGHTS
+    if (VarGet(gSpecialVar_0x8003) == 1)
+        return GetSpecialBattleTransition(FRONTIER_MODE_DOUBLES);
+    #endif
```

Comment:
the preproc block should be a normal if statement

Reply:
I'll gladly make the change but could you please tell me why that is the preferred way?

Reply:
because we can't verify if the code compiles with a preproc if it is set to false.

Reply:
That totally makes sense, thank you for the explanation.